frontend/test/metabase/lib/expressions/typechecker.unit.spec.js

```diff
@@ -0,0 +1,136 @@
+import { parse } from "metabase/lib/expressions/parser";
+import { ExpressionVisitor } from "metabase/lib/expressions/visitor";
+import { parseIdentifierString } from "metabase/lib/expressions/index";
+
+describe("type-checker", () => {
+  function parseSource(source, startRule) {
+    let cst = null;
+    try {
+      cst = parse({ source, tokenVector: null, startRule }).cst;
+    } catch (e) {
+      let err = e;
+      if (err.length && err.length > 0) {
+        err = err[0];
+        if (typeof err.message === "string") {
+          err = err.message;
+        }
+      }
+      throw err;
+    }
```

Comment:
Is this used? It seems like checking the thrown exceptions would be good.

Reply:
This is just to simplify the exceptions produced by Chevrotain (the exception will be still thrown again).
And yes, once the type checker is a better shape, I plan to add invalid test cases which will trigger this exception.